### ISSUE TITLE
phpcs doesn't recognize docblock templates

### ISSUE THREAD

**Issue by jpuck** [squizlabs/PHP_CodeSniffer#2014]
Please enlighten me if I'm missing a configuration, but phpcs doesn't seem to recognize [docblock templates][docref].

For example,

```php
/**#@+
 * @const string
 */
const EQUAL = '==';
const NOT_EQUAL = '!=';
const GREATER_THAN = '>';
const LESS_THAN = '<';
const GREATER_THAN_EQUAL = '>=';
const LESS_THAN_EQUAL = '<=';
/**#@-*/
```

will return this error:

```
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------
 56 | ERROR | [x] Expected 1 blank line(s) before member var; 2 found (Squiz.WhiteSpace.MemberVarSpacing.Incorrect)
--------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
```

but to fix it would remove `/**#@-*/` because it thinks it's a blank line.

Thanks in advance for your assistance, and all your hard work on this project!

[docref]:https://en.wikipedia.org/wiki/PHPDoc#DocBlock_Templates

**Comment by gsherwood** [comment]
No, none of the included PHPCS sniffs have any support for docblock templates and it's pretty unlikely they will get support unless someone else takes on the job as my todo list is really long. Sorry for the bad news.

**Comment by gsherwood** [comment]
I don't know anything about docblock templates, but:

> What would be a recommended approach to implementing it?

Maybe something in the comment tokenizer. It might require a different comment token type and then changes to pretty much all the commenting sniffs.

> Are there some similar examples you can think of?

I can't think of any.

> Even if not me, then documenting your thoughts here might be helpful for anyone else who might pick up this task in the future.

I really don't have time to research and plan this task, so I suggest that anyone wanting to help might be best off doing that first. I may take a while to respond, but I do read all the comments and may be able to point people in the right direction. But my experience with commenting standards leads me to think this may end up being more complex than it seems. But again, I haven't researched it, so I'm just guessing based on past experience.

**Comment by jpuck** [comment]
for anyone looking for a workaround on this issue, you can explicitly ignore the specific rule `phpcs:ignore Squiz.WhiteSpace.MemberVarSpacing.Incorrect` on the line of the first property.

```php
/**#@+
 * @const string Comparison constants
 */
const EQUAL = '==';
const NOT_EQUAL = '!=';
const GREATER_THAN = '>';
const LESS_THAN = '<';
const GREATER_THAN_EQUAL = '>=';
const LESS_THAN_EQUAL = '<=';
/**#@-*/

/**
 * @var string Some value
 */
protected $foo = 'bar'; // phpcs:ignore Squiz.WhiteSpace.MemberVarSpacing.Incorrect
```

**Comment by rodrigoprimo** [comment]
Please correct if I'm missing something, but I believe that DocBlock templates were a phpDocumentor feature that was never carried over to phpDocumentor 2. The phpDocumentor maintainers explicitly decided not to reimplement the feature due to its complexity and low demand (phpDocumentor/phpDocumentor#42, phpDocumentor/phpDocumentor#2225).                                                                                                                                                                                                                                                         

Since the feature has been unsupported by phpDocumentor for several years, maybe this issue can be closed? Or is there still a reason to support docblock templates in PHPCS?
